### PR TITLE
#162727473 User should be able to follow artists

### PR DESCRIPTION
**Pull request by daniellamarr**
#### What does this PR do?
- This PR allows users to follow artists
- This PR allows artists to follow other artists
#### Description of Task to be completed?
- Authenticated users should be able to follow artists
- Authenticated artists should be able to follow other artists
- There should be a summary of user followers and following
- Users should see a list of users they follow
- Users should see a list of users that follow them
#### How should this be manually tested?
- Run `npm i`
- Send a **POST** request to `/api/v1/users/artists/follow/:artistId` to follow artist, you must be logged in and a token must be passed to the request before you can access the route. `artistId` represents the ID of a particular artist
- Send a **POST** request to `/api/v1/users/artists/unfollow/:artistId`, where `artistId` is the artist Id. Pass in access token before accessing the route. The route gives authorized users access to unfollow an artist
- Send a **GET** request to `/api/v1/users/:userId/followers`, where `userId` is the user's id. Pass in access token before accessing the route. The route returns a list of users that follow another user in reference to the `userId`.
- Send a **GET** request to `/api/v1/users/:userId/following`, where `userId` is the user's id. Pass in access token before accessing the route. The route returns a list of users that a user follows in reference to the `userId`
#### Any background context you want to provide?
- The user must have signed up and logged in on the app before attempting to test this feature.
#### What are the relevant pivotal tracker stories?
[#162727473](https://www.pivotaltracker.com/story/show/162727473)
#### Screenshots (if appropriate)
screen shot 2019-01-17 at 9 14 07 pm 2
screen shot 2019-01-17 at 9 16 40 pm 2

#### Questions:
N/A

### CODE REVIEW

File: src/controllers/UsersController.js [hunk 1]
```diff
@@ -4,9 +4,11 @@ import TokenAuthenticate from '../helpers/TokenAuthenticate';
 import Response from '../helpers/response';
 import { passwordHash, comparePassword } from '../helpers/passwordHash';
 import EmailNotificationAPI from '../helpers/EmailNotificationAPI';
-import basePath from '../helpers/basePath';
+import basePath from '../helpers/basepath';
```
Comment (julietezekwe) [inline src/controllers/UsersController.js:7]: Please, can you make the import name and filename same?  camelCase preferably.

File: src/db/service/Unfollow.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  static async unfollower(id) {
+    const findFollowingSummary = await FollowSummary.findOne({
+      where: {
+        userId: id,
+      }
+    });
+    if (findFollowingSummary) {
+      const following = findFollowingSummary.following - 1;
+      findFollowingSummary.following = following;
+      findFollowingSummary.save();
```
Comment (abejide001) [inline src/db/service/Unfollow.js:25]: can you explain this line?

**Review by abejide001 — CHANGES_REQUESTED** [review]

File: src/routes/userRouter.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+userRouter.post(
+  '/artists/follow/:artistId',
+  TokenAuthenticate.tokenVerify,
+  UserMiddleware.validateFollowerAndArtistID,
+  UsersController.userFollow,
+);
+
+userRouter.post(
+  '/artists/unfollow/:artistId',
+  TokenAuthenticate.tokenVerify,
```
Comment (Proception) [inline src/routes/userRouter.js:18]: Please can you take `TokenAuthenticate.tokenVerify` to the index router since all routes here are protected

**Review by Proception — COMMENTED** [review]
Please implement feedback given

File: src/controllers/UsersController.js (hunk 1 above)
Comment (daniellamarr) [reply src/controllers/UsersController.js:7]: Alright

**Review by wombolo — APPROVED** [review]
LGTM

File: src/db/service/Unfollow.js (hunk 2 above)
Comment (daniellamarr) [reply src/db/service/Unfollow.js:25]: This line simply updates the current number of people a user follows in the summary table. So, the `save` is an accessor function on `sequelize` that immediately updates the specified table.

Take note of the `following` variable passed into the `findFollowingSummary` object. The variable holds the number of artists a user currently follows.

**Review by abejide001 — APPROVED** [review]

File: src/controllers/UsersController.js [hunk 4]
```diff
@@ -423,6 +525,105 @@ class UsersController {
...
+   * @param {object} res
+   * @memberof UsersController
+   * @returns all users following the authenticated user
+   */
+  static async getFollowers(req, res) {
+    try {
+      const { followers } = req;
+      /* eslint-disable prefer-const  */
+      let queries = [];
+      followers.forEach((x) => {
```
Comment (andela-dbamidele) [inline src/controllers/UsersController.js]: Is there a better way to do this?

**Review by julietezekwe — APPROVED** [review]

File: src/controllers/UsersController.js (hunk 4 above)
Comment (andela-dbamidele) [reply src/controllers/UsersController.js]: You haven't updated this yet....

Comment (daniellamarr) [reply src/controllers/UsersController.js]: I have done it now.

**Review by andela-dbamidele — APPROVED** [review]

**Pull request merged**